src/os/windows/crashlog_win.cpp

```diff
@@ -462,44 +462,51 @@ static INT_PTR CALLBACK CrashDialogFunc(HWND wnd, UINT msg, WPARAM wParam, LPARA
 {
 	switch (msg) {
 		case WM_INITDIALOG: {
+			size_t crashlog_length = CrashLogWindows::current->crashlog.size() + 1;
+			/* Reserve extra space for LF to CRLF conversion. */
+			crashlog_length += std::count(CrashLogWindows::current->crashlog.begin(), CrashLogWindows::current->crashlog.end(), '\n');
+
+			const size_t filename_count = 4;
+			const size_t filename_buf_length = MAX_PATH + 1;
+			const size_t crash_desc_buf_length = lengthof(_crash_desc) + filename_buf_length * filename_count + 1;
+
 			/* We need to put the crash-log in a separate buffer because the default
-			 * buffer in MB_TO_WIDE is not large enough (512 chars) */
-			wchar_t filenamebuf[MAX_PATH * 2];
-			wchar_t crash_msgW[65536];
-			/* Convert unix -> dos newlines because the edit box only supports that properly :( */
-			const char *unix_nl = CrashLogWindows::current->crashlog.data();
-			char dos_nl[65536];
-			char *p = dos_nl;
+			 * buffer in MB_TO_WIDE is not large enough (512 chars).
+			 * Use VirtualAlloc to allocate pages for the buffer to avoid overflowing the stack.
+			 * Avoid the heap in case the crash is because the heap became corrupted. */
+			const size_t total_length = crash_desc_buf_length * sizeof(wchar_t) + \
```

Comment:
`\` at end of line are only for preprocessor stuff like macros, the compiler knows the C++ line ends at the next `;`